Updated the grammar rules to allow for optional trailing commas.

MahtabNorouzi opened the pull request:
<!-- Please ensure that your PR includes the following, as needed -->

- [ ] Tests added for any new code
- [ ] Documentation added for any new functionality
- [ ] Entries added to the respective `CHANGELOG.md` for any new functionality
- [ ] Feature table on [`README.md`](../README.md#roadmap) updated for any listed functionality

<!--
Some common CI checks and how to fix them (if failing):
- The formatting in all files is consistent with the project's style.
   - Run `npm run format` to automatically format all files.
- The `examples/README.md` file contains all Quint files in `examples/`
  and correctly lists their ability to go through pipeline stages.
   - Run `make examples` to automatically regenerate this file locally.
- The assets in `quint/testFixture` and `doc/builtin.md` are consistent.
   - Run `npm run generate` to automatically update these files locally.
-->

bugarela commented on quint/src/generated/Quint.g4:
```diff
@@ -92,7 +92,7 @@ exportMod : 'export' name '.' identOrStar
+                'import' moduleName '(' (name '=' expr (',' name '=' expr)*) ','? ')' '.' '*'
                   ('from' fromSource)?
                 // creating an instance and importing all names with a prefix
             |   'import' moduleName '(' (name '=' expr (',' name '=' expr)*) ')' 'as' qualifiedName
```
We should also allow the comma on this form to be consistent

bugarela reviewed (commented):
We should add a test for this, such as adding some usages of all new forms of trailing commas to `SuperSpec.qnt` - and then updating `SuperSpec.json` by running `npm run update-fixtures`

MahtabNorouzi commented:
## Summary of Changes
- Updated the grammar rules to allow for optional trailing commas.
- Modified the `min` function in `superspec.map.json` to demonstrate this change while keeping the `add` function unchanged for reference.
- Added an import statement: `import Proto(N = N,) as Inst1`.

## Details
- The `min` function now accepts parameters in the following formats:
  - `(x: int, y: int,)`
  - `(x: int, y: int)`
- The import statement is included to utilize the `Proto` module with an optional trailing comma.

## Reason for Changes
- Allowing an optional trailing comma makes the syntax more flexible and consistent with common programming practices.

## Test Cases
- Added test cases for both scenarios in `superspec.map.json` to ensure correct parsing of functions with optional trailing commas.

bugarela reviewed (approved):
Looks great! Before merging it, can you add "Closes #1383" to the PR description, so the issue gets automatically closed when we merge this?

bugarela commented:
Actually, I remembered something else: we should add an entry to CHANGELOG.md saying that we now allow trailing commas in operator definitions and constant initialization - which also makes me think we might have missed one case: operator call. We should be able to call `min(1, 2,)` just as well as we can define `min(x: int, y: int,)`

bugarela commented on CHANGELOG.md:
```diff
@@ -14,6 +14,25 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
+
+- Updated grammar rule to allow an optional trailing comma in parameter lists:
+  
+  - Function calls
```
```suggestion
  - Operator calls
```

bugarela commented on CHANGELOG.md:
```diff
@@ -12,11 +12,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
+
+- Updated grammar rule to allow an optional trailing comma in parameter lists:
```
It's good to add a reference to an issue/PR for whoever needs more context and details.
```suggestion
- Updated grammar rule to allow an optional trailing comma in parameter lists (#1510):
```

bugarela reviewed (approved):
Looks great! I've just merged another PR that touches the grammar, so you'll need to re-generate the files once merging main on this - sorry about that! Otherwise, we're good to go :sparkles:
